src/main/java/org/mule/extension/ftp/api/ftp/FtpFileAttributes.java

```diff
@@ -44,130 +44,196 @@ public class FtpFileAttributes extends AbstractFileAttributes {
   @Parameter
   private boolean symbolicLink;
 
+  @Parameter
+  protected final String path;
+
+  @Parameter
+  private final String fileName;
+
   /**
-   * Creates a new instance
+   * Creates a new instance of FtpFileAttributes with the specified URI and FTPFile.
    *
-   * @param uri the file's {@link URI}
+   * @param uri     the file's {@link URI}
    * @param ftpFile the {@link FTPFile} which represents the file on the FTP server
    */
-
-
-
   public FtpFileAttributes(URI uri, FTPFile ftpFile) {
-    super(uri);
+    this.path = uri.getPath();
+    String name = FilenameUtils.getName(uri.getPath());
+    this.fileName = name != null ? name : "";
     timestamp = ftpFile.getTimestamp() != null ? asDateTime(ftpFile.getTimestamp().toInstant()) : null;
-    // TODO MULE-15337: Remove redundant 'name' attribute in next major version
-    name = ftpFile.getName() != null ? ftpFile.getName() : "";
     size = ftpFile.getSize();
     regularFile = ftpFile.isFile();
     directory = ftpFile.isDirectory();
     symbolicLink = ftpFile.isSymbolicLink();
   }
 
-
+  /**
+   * Creates a new instance of FtpFileAttributes with default values.
+   */
   public FtpFileAttributes() {
-    super(createDefaultUri());
+    URI uri = createDefaultUri();
+    this.path = uri.getPath();
+    String name = FilenameUtils.getName(uri.getPath());
+    this.fileName = name != null ? name : "";
```

Comment:
What's wrong with having the fileName `null` instead of empty `""`?

Reply:
This is an existing code which is moved while refactoring code from AbstractFileAttributes to FtpFileAttributes. Changing this null here will send value as null and may cause issue to customer.